### PR TITLE
mark get_event_records as deprecated

### PR DESCRIPTION
**Pull request by prha**
## Summary & Motivation
We want to restrict the type of events we support fetching records for, for performance sake.

## How I Tested These Changes
Inspection

### CODE REVIEW

File: python_modules/dagster/dagster/_core/instance/__init__.py [hunk 1]
```diff
@@ -1988,8 +1988,8 @@ def get_latest_asset_check_evaluation_record(
             asset_check_key
         )
 
-    @public
     @traced
+    @deprecated(breaking_version="2.0")
```
Comment (sryza) [inline python_modules/dagster/dagster/_core/instance/__init__.py:1992]: Should we direct people to "use `fetch_materializations` or an equivalent method instead"?

Comment (prha) [reply python_modules/dagster/dagster/_core/instance/__init__.py:1992]: I think we could if we know which events they were trying to fetch.

If they're using them to fetch step events, there is not a clear equivalent method.

I'm debating whether or not to mark deprecated, or just remove the public designation.

Comment (sryza) [reply python_modules/dagster/dagster/_core/instance/__init__.py:1992]: I was imagining we would just provide a generic message that mentions several fetch methods, not necessarily tailored to the event type they're searching for.

To your point though, we could tell from the filter in many cases, right? And in the case where no filter is provided, tell them that getting records across event types is deprecated with no replacement?

I think we generally avoid removing the public label from non-experimental methods once we've added it. Public is kind of the method equivalent of exporting a symbol at the top level. If we did take off the public label, we'd need to somehow indicate to ourselves that we can't remove it until 2.0. And if we want users to move away from it, then I think raising a warning is good.

**Review by schrockn — COMMENTED** [review]
I do think we should

1) Mark as deprecated
2) Provide explanatory text as to the reasoning and what the alternatives are.

**Review by schrockn — CHANGES_REQUESTED** [review]
Let's provide explanatory text.

File: python_modules/dagster/dagster/_core/instance/__init__.py (hunk 1 above)
Comment (schrockn) [reply python_modules/dagster/dagster/_core/instance/__init__.py:1992]: Let's update this additional warn text telling users what they should be using instead.

**Review by schrockn — CHANGES_REQUESTED** [review]
Better warning text please! Otherwise strongly support.

**Review by schrockn — APPROVED** [review]
Oh whoops I was looking forward in the text of the decorator

**Pull request merged**